apps/drec-api/src/pods/reads/reads.service.ts

```diff
@@ -0,0 +1,103 @@
+import { Inject, Injectable, Logger, NotFoundException } from '@nestjs/common';
+import {
+  MeasurementDTO,
+  ReadsService as BaseReadService,
+  Unit,
+} from '@energyweb/energy-api-influxdb';
+import { DeviceService } from '../device/device.service';
+import { OrganizationService } from '../organization/organization.service';
+import moment from 'moment';
+import { Device } from '../device/device.entity';
+import { BASE_READ_SERVICE } from './const';
+import { EventBus } from '@nestjs/cqrs';
+import { GenerationReadingStoredEvent } from '../../events/GenerationReadingStored.event';
+import { BigNumber } from 'ethers';
+
+@Injectable()
+export class ReadsService {
+  private readonly logger = new Logger(ReadsService.name);
+
+  constructor(
+    @Inject(BASE_READ_SERVICE)
+    private baseReadsService: BaseReadService,
+    private readonly deviceService: DeviceService,
+    private readonly organizationService: OrganizationService,
+    private readonly eventBus: EventBus,
+  ) {}
+
+  public async storeRead(
+    id: string,
+    measurements: MeasurementDTO,
+  ): Promise<void> {
+    this.logger.debug('DREC is storing smart meter reads:');
+    this.logger.debug(JSON.stringify(measurements));
+
+    const device = await this.deviceService.findOne(+id);
+    const roundedMeasurements = this.roundMeasurementsToUnit(measurements);
+    await this.storeGenerationReading(id, roundedMeasurements, device);
+  }
+
+  private async storeGenerationReading(
+    id: string,
+    measurements: MeasurementDTO,
+    device: Device,
+  ): Promise<void> {
+    const organization = await this.organizationService.findOne(
+      device.registrant_organisation_code,
+    );
+
+    if (!organization) {
+      throw new NotFoundException(
+        `No organization found with device organization code ${device.registrant_organisation_code}`,
+      );
+    }
+
+    await this.store(id, measurements);
+
+    for (const measurement of measurements.reads) {
+      const startTime = moment(measurement.timestamp)
+        .subtract(30, 'minute')
+        .toDate();
+      const endTime = moment(measurement.timestamp).toDate();
+
+      this.eventBus.publish(
+        new GenerationReadingStoredEvent({
```

Comment:
Please import this event from sdk transfer instead of local one. You don't have to import transfer module into application if you don't want yet.

Or is it completely another project, and you won't use transfer module?

Reply:
It`s another project and I don`t think we would need origin-247-transfer' as it is not a 24-7 project.